source/dub/internal/vibecompat/inet/path.d

```diff
@@ -283,7 +283,7 @@ struct PathEntry {
 
 	string toString() const { return m_name; }
 
-	Path opBinary(string OP)(PathEntry rhs) const if( OP == "~" ) { return Path(cast(immutable)[this, rhs], false); }
+	Path opBinary(string OP)(PathEntry rhs) const if( OP == "~" ) { return Path(([this, rhs]).idup, false); }
```

Comment:
The casting changes aren't part of the fix, but the possibility of undefined behavior convinced me to keep it in my patches.
That being said, I dont think it will be an issue unless another part of the code tries to change the memory. So it should be fine. 
The main issue was the inplace reassignment (and only in certain versions of glibc, I think.)
If you pull,  feel free to change the immutable casts as you want. 
I'd do it myself but I'm indisposed with real life stuff today (moving to a new home) :p

-------- Original message --------
From: Sönke Ludwig <notifications@github.com> 
Date: 07/17/2015  6:45 AM  (GMT-05:00) 
To: D-Programming-Language/dub <dub@noreply.github.com> 
Cc: bittwiddler1 <beniquezsd@outlook.com> 
Subject: Re: [dub] Bugfix for #601. Inplace assignment of 'p =
  p.relativeTo(cwd)' was (#625) 

In source/dub/internal/vibecompat/inet/path.d:
> @@ -283,7 +283,7 @@ struct PathEntry {
>  
>  	string toString() const { return m_name; }
>  
> -	Path opBinary(string OP)(PathEntry rhs) const if( OP == "~" ) { return Path(cast(immutable)[this, rhs], false); }
> +	Path opBinary(string OP)(PathEntry rhs) const if( OP == "~" ) { return Path(([this, rhs]).idup, false); }

Same here, the cast can simply be be dropped for all DMD versions down to 2.064 as it seems.

—
Reply to this email directly or view it on GitHub.